lib/bibliothecary/parsers/nuget/nuget_range.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Bibliothecary
+  module Parsers
+    class Nuget
+      class NugetRange
+        # Regular expressions for different NuGet version range patterns
+        MINOR_WILDCARD_REGEXP = /^(\d+)\.\*$/                         # 1.* - minor wildcard
+        PATCH_WILDCARD_REGEXP = /^(\d+)\.(\d+)\.\*$/                  # 1.2.* - patch wildcard
+        FULL_WILDCARD_REGEXP = /^\*$/                                 # * - full wildcard
+        MINIMUM_VERSION_REGEXP = /^\d+\.\d+.*$/                       # Minimum version like "1.0"
+        INCLUSIVE_LOWER_BOUND_REGEXP = /^\[([^,\]]+),\)$/             # [1.0,) - inclusive lower bound, no upper
+        EXCLUSIVE_LOWER_BOUND_REGEXP = /^\(([^,\]]+),\)$/             # (1.0,) - exclusive lower bound, no upper
+        EXACT_VERSION_REGEXP = /^\[([^,\]]+)\]$/                      # [1.0] - exact version
+        INCLUSIVE_UPPER_BOUND_REGEXP = /^\(,([^,\]]+)\]$/             # (,1.0] - no lower, inclusive upper bound
+        EXCLUSIVE_UPPER_BOUND_REGEXP = /^\(,([^,\]]+)\)$/             # (,1.0) - no lower, exclusive upper bound
+        INCLUSIVE_RANGE_REGEXP = /^\[([^,\]]+),([^,\]]+)\]$/          # [1.0,2.0] - inclusive range
+        EXCLUSIVE_RANGE_REGEXP = /^\(([^,\]]+),([^,\]]+)\)$/          # (1.0,2.0) - exclusive range
+        INCLUSIVE_LOWER_EXCLUSIVE_UPPER_REGEXP = /^\[([^,\]]+),([^,\]]+)\)$/ # [1.0,2.0) - inclusive lower, exclusive upper
+        EXCLUSIVE_LOWER_INCLUSIVE_UPPER_REGEXP = /^\(([^,\]]+),([^,\]]+)\]$/ # (1.0,2.0] - exclusive lower, inclusive upper
+
+        # Converts NuGet version range syntax to a SemVer requirement syntax
+        # supported by the semantic_range gem (https://github.com/librariesio/semantic_range).
+        # Nuget docs: https://learn.microsoft.com/en-us/nuget/concepts/package-versioning#version-ranges
+        #
+        # @param requirement [String] The NuGet version range string
+        # @return [String] The converted SemVer requirement string
+        # @raise [ArgumentError] If the requirement format is invalid
+        def self.convert_to_semver_requirement(requirement)
+          case requirement
+          when MINOR_WILDCARD_REGEXP
+            "~> #{::Regexp.last_match(1)}.0"
+          when PATCH_WILDCARD_REGEXP
+            ">= #{::Regexp.last_match(1)}.0"
+          when FULL_WILDCARD_REGEXP
+            ">= 0"
```

Comment:
I'm on the fence about this: we've solidified around a default "*" wildcard for most Bibliothecary parsers, which is effectively the same as ">= 0".

So returning ">= 0" is more semantically consistent, but "*" is also safe and would be more consistent with Bibliothecary.

Thoughts?